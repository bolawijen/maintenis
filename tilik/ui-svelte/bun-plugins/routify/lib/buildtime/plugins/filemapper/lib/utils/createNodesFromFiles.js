import fse from 'fs-extra'
import { relative, resolve } from 'path'
import { File } from '../File.js'

const isInWantedExtensions = (extensions, file) =>
    extensions.find(ext =>
        typeof ext === 'string' ? file.path.endsWith(ext) : ext.test(file.path),
    )

/**
 * Maps filestructure to a node tree
 * @param {RNodeBuildtime} firstNode the root node of the specified path
 * @param {String} path dir to scan for files
 * @return {Promise<RNodeBuildtime>}
 */
export async function createNodesFromFiles(firstNode, path) {
    firstNode.file = new File(path)
    const queue = [firstNode]

    let node

    while ((node = queue.pop())) {
        const relativePath = relative(node.root.file.path, node.file.path)
        // set id to `default_path_to_file` if rootName is default
        node.id =
            '_' +
            /** @ts-ignore rootName doesn't exist */
            [node.root.rootName, relativePath]
                .filter(Boolean)
                .join('_')
                .replace(/[^\w]/g, '_')

        if (node.file.stat.isDirectory()) {
            // for each child create a node, attach a file and add it to the queue
            const children = await fse.readdir(node.file.path)
            const promises = children.map(filename => {
                const file = new File(resolve(node.file.path, filename))
                if (
                    (file.stat.isDirectory() ||
                        isInWantedExtensions(
                            firstNode.instance.options.extensions,
                            file,
                        )) &&
                    !file.name.startsWith('__') &&
                    !file.name.startsWith('.')
                ) {
                    const module = !file.stat.isDirectory() && file.path
                    const childNode = node.createChild(file.name, module)
                    childNode.file = file
                    queue.unshift(childNode)
                }
            })
            await Promise.all(promises)
        }
    }
    return firstNode
}
