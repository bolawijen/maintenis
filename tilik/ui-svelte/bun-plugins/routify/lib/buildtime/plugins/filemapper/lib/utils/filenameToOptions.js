import { nameFilter } from '../../utils.js'

/**
 * assign reset, fallback and dynamic paths to node.meta
 * @param {RNodeBuildtime} node rootNode of the instance
 */
export const filenameToOptions = node => {
    const { descendants } = node

    const { extensions } = node.instance.options
    const { resetFiles, fallbackFiles } = node.instance.options.filemapper

    const resetNames = resetFiles.map(name => extensions.map(ext => name + ext)).flat()
    const fallbackNames = fallbackFiles
        .map(name => extensions.map(ext => name + ext))
        .flat()

    const indexNames = extensions.map(ext => 'index' + ext)

    descendants
        .filter(nameFilter(indexNames))
        .forEach(node => (node.meta.isDefault = true))

    // set meta.reset for _reset.svelte
    descendants.filter(nameFilter(resetNames)).forEach(node => (node.meta.reset = true))

    // set meta.fallback for _fallback.svelte
    descendants.filter(nameFilter(fallbackNames)).forEach(node => {
        node.meta.fallback = true
        node.meta.order = false
    })

    // set meta.dynamic for files with attribute in name
    descendants
        .filter(node => node.file && node.file.name.match(/\[.+\]/))
        .forEach(node => {
            node.meta.dynamic = true
            node.meta.order = false
        })

    // set meta.dynamicSpread for files with attribute in name
    descendants
        .filter(node => node.file && node.file.name.match(/\[\.\.\..+\]/))
        .forEach(node => {
            node.meta.dynamicSpread = true
            node.meta.order = false
        })

    descendants
        .filter(
            node =>
                node.name.startsWith('_') && !node.name.match(/_(module|layout|reset)/),
        )
        .forEach(node => {
            node.meta.noRoute = true
        })
    descendants
        .filter(node => node.name === '_decorator')
        .forEach(node => {
            node.meta.isDecorator = true
        })
    descendants
        .filter(node => node.name === '_inlineWrapper')
        .forEach(node => {
            node.meta.isInlineWrapper = true
        })
}
