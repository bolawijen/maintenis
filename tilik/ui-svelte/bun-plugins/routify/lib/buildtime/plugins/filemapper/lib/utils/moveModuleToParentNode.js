import { nameFilter } from '../../utils.js'

/**
 * for nodes that have a _module.svelte or _reset.svelte file,
 * `file` prop and module is moved to node.parent and the old node is removed
 * @param {RNodeBuildtime} node
 */
export const moveModuleToParentNode = node => {
    const { options } = node.instance
    const { moduleFiles } = options.filemapper
    const { extensions } = options

    const filenames = moduleFiles.map(name => extensions.map(ext => name + ext)).flat()

    node.descendants.filter(nameFilter(filenames)).forEach(node => {
        node.parent.module = node.module
        node.parent.file = node.file
        node.parent.meta = { ...node.parent.meta, ...node.meta }
        node.remove()
    })
}
