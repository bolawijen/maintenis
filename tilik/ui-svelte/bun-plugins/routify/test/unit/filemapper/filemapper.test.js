import { dirname } from 'path'
import { fileURLToPath } from 'url'
import { createNodesFromFiles } from '../../../lib/buildtime/plugins/filemapper/lib/utils/createNodesFromFiles.js'
import { moveModuleToParentNode } from '../../../lib/buildtime/plugins/filemapper/lib/utils/moveModuleToParentNode.js'
import { filenameToOptions } from '../../../lib/buildtime/plugins/filemapper/lib/utils/filenameToOptions.js'
import { filemapper } from '../../../lib/buildtime/plugins/filemapper/lib/index.js'
import { RoutifyBuildtime } from '../../../lib/buildtime/RoutifyBuildtime.js'

const __dirname = dirname(fileURLToPath(import.meta.url))

const options = {
    routesDir: { default: `${__dirname}/example/routes` },
    filemapper: {
        moduleFiles: ['_module.svelte', '_reset.svelte'],
        resetFiles: ['_reset.svelte'],
    },
}

const instance = new RoutifyBuildtime(options)

let rootNode = instance.createNode()
rootNode.rootName = 'default'

test('files are mapped', async () => {
    await createNodesFromFiles(rootNode, options.routesDir.default)
    expect(instance.nodeIndex.length).toBe(12)
    expect(rootNode).toMatchSnapshot('1.filemap-only')
})

test('modules are merged with parent node', async () => {
    moveModuleToParentNode(rootNode)
    expect(instance.nodeIndex.length).toBe(10)
    expect(rootNode).toMatchSnapshot('2.filemap-with-modules')
})

test('options get added', async () => {
    filenameToOptions(rootNode)
    expect(rootNode).toMatchSnapshot('3.filemap-with-resets')
})

test('filemapper', async () => {
    const instance = new RoutifyBuildtime(options)
    await filemapper({ instance })
    expect(Object.values(instance.rootNodes)[0]).toMatchSnapshot(
        '4.filemap-with-components',
    )
})
