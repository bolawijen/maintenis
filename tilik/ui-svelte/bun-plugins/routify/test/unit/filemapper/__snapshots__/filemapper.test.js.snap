// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/unit/filemapper/filemapper.test.js/filemapper: 4.filemap-with-components 1`] = `
Object {
  "asyncModule": undefined,
  "children": Array [
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "_reset.svelte",
        "dir": "test/unit/filemapper/example/routes/admin",
        "ext": ".svelte",
        "name": "_reset",
        "path": "test/unit/filemapper/example/routes/admin/_reset.svelte",
      },
      "id": "_default_admin",
      "meta": Object {
        "reset": true,
      },
      "module": "test/unit/filemapper/example/routes/admin/_reset.svelte",
      "name": "admin",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "[slug].svelte",
            "dir": "test/unit/filemapper/example/routes/blog",
            "ext": ".svelte",
            "name": "[slug]",
            "path": "test/unit/filemapper/example/routes/blog/[slug].svelte",
          },
          "id": "_default_blog__slug__svelte",
          "meta": Object {
            "dynamic": true,
            "order": false,
          },
          "module": "test/unit/filemapper/example/routes/blog/[slug].svelte",
          "name": "[slug]",
        },
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "prefixed[slug].svelte",
            "dir": "test/unit/filemapper/example/routes/blog",
            "ext": ".svelte",
            "name": "prefixed[slug]",
            "path": "test/unit/filemapper/example/routes/blog/prefixed[slug].svelte",
          },
          "id": "_default_blog_prefixed_slug__svelte",
          "meta": Object {
            "dynamic": true,
            "order": false,
          },
          "module": "test/unit/filemapper/example/routes/blog/prefixed[slug].svelte",
          "name": "prefixed[slug]",
        },
      ],
      "file": File {
        "base": "blog",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "blog",
        "path": "test/unit/filemapper/example/routes/blog",
      },
      "id": "_default_blog",
      "meta": Object {},
      "module": false,
      "name": "blog",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "index.svelte",
        "dir": "test/unit/filemapper/example/routes",
        "ext": ".svelte",
        "name": "index",
        "path": "test/unit/filemapper/example/routes/index.svelte",
      },
      "id": "_default_index_svelte",
      "meta": Object {
        "isDefault": true,
      },
      "module": "test/unit/filemapper/example/routes/index.svelte",
      "name": "index",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [
            Object {
              "asyncModule": undefined,
              "children": Array [],
              "file": File {
                "base": "foo.svelte",
                "dir": "test/unit/filemapper/example/routes/no-layouts/nested",
                "ext": ".svelte",
                "name": "foo",
                "path": "test/unit/filemapper/example/routes/no-layouts/nested/foo.svelte",
              },
              "id": "_default_no_layouts_nested_foo_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/routes/no-layouts/nested/foo.svelte",
              "name": "foo",
            },
            Object {
              "asyncModule": undefined,
              "children": Array [],
              "file": File {
                "base": "index.svelte",
                "dir": "test/unit/filemapper/example/routes/no-layouts/nested",
                "ext": ".svelte",
                "name": "index",
                "path": "test/unit/filemapper/example/routes/no-layouts/nested/index.svelte",
              },
              "id": "_default_no_layouts_nested_index_svelte",
              "meta": Object {
                "isDefault": true,
              },
              "module": "test/unit/filemapper/example/routes/no-layouts/nested/index.svelte",
              "name": "index",
            },
          ],
          "file": File {
            "base": "nested",
            "dir": "test/unit/filemapper/example/routes/no-layouts",
            "ext": "",
            "name": "nested",
            "path": "test/unit/filemapper/example/routes/no-layouts/nested",
          },
          "id": "_default_no_layouts_nested",
          "meta": Object {},
          "module": false,
          "name": "nested",
        },
      ],
      "file": File {
        "base": "no-layouts",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "no-layouts",
        "path": "test/unit/filemapper/example/routes/no-layouts",
      },
      "id": "_default_no_layouts",
      "meta": Object {},
      "module": false,
      "name": "no-layouts",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "[...404].svelte",
        "dir": ".routify/components",
        "ext": ".svelte",
        "name": "[...404]",
        "path": ".routify/components/[...404].svelte",
      },
      "id": undefined,
      "meta": Object {
        "dynamic": true,
        "dynamicSpread": true,
        "order": false,
      },
      "module": "./.routify/components/[...404].svelte",
      "name": "[...404]",
    },
  ],
  "file": File {
    "base": "_module.svelte",
    "dir": "test/unit/filemapper/example/routes",
    "ext": ".svelte",
    "name": "_module",
    "path": "test/unit/filemapper/example/routes/_module.svelte",
  },
  "id": "_default",
  "meta": Object {},
  "module": "test/unit/filemapper/example/routes/_module.svelte",
  "name": "",
  "rootName": "default",
}
`;

exports[`test/unit/filemapper/filemapper.test.js/files are mapped: 1.filemap-only 1`] = `
Object {
  "asyncModule": undefined,
  "children": Array [
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "_module.svelte",
        "dir": "test/unit/filemapper/example/routes",
        "ext": ".svelte",
        "name": "_module",
        "path": "test/unit/filemapper/example/routes/_module.svelte",
      },
      "id": "_default__module_svelte",
      "meta": Object {},
      "module": "test/unit/filemapper/example/routes/_module.svelte",
      "name": "_module",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "_reset.svelte",
            "dir": "test/unit/filemapper/example/routes/admin",
            "ext": ".svelte",
            "name": "_reset",
            "path": "test/unit/filemapper/example/routes/admin/_reset.svelte",
          },
          "id": "_default_admin__reset_svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/routes/admin/_reset.svelte",
          "name": "_reset",
        },
      ],
      "file": File {
        "base": "admin",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "admin",
        "path": "test/unit/filemapper/example/routes/admin",
      },
      "id": "_default_admin",
      "meta": Object {},
      "module": false,
      "name": "admin",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "[slug].svelte",
            "dir": "test/unit/filemapper/example/routes/blog",
            "ext": ".svelte",
            "name": "[slug]",
            "path": "test/unit/filemapper/example/routes/blog/[slug].svelte",
          },
          "id": "_default_blog__slug__svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/routes/blog/[slug].svelte",
          "name": "[slug]",
        },
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "prefixed[slug].svelte",
            "dir": "test/unit/filemapper/example/routes/blog",
            "ext": ".svelte",
            "name": "prefixed[slug]",
            "path": "test/unit/filemapper/example/routes/blog/prefixed[slug].svelte",
          },
          "id": "_default_blog_prefixed_slug__svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/routes/blog/prefixed[slug].svelte",
          "name": "prefixed[slug]",
        },
      ],
      "file": File {
        "base": "blog",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "blog",
        "path": "test/unit/filemapper/example/routes/blog",
      },
      "id": "_default_blog",
      "meta": Object {},
      "module": false,
      "name": "blog",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "index.svelte",
        "dir": "test/unit/filemapper/example/routes",
        "ext": ".svelte",
        "name": "index",
        "path": "test/unit/filemapper/example/routes/index.svelte",
      },
      "id": "_default_index_svelte",
      "meta": Object {},
      "module": "test/unit/filemapper/example/routes/index.svelte",
      "name": "index",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [
            Object {
              "asyncModule": undefined,
              "children": Array [],
              "file": File {
                "base": "foo.svelte",
                "dir": "test/unit/filemapper/example/routes/no-layouts/nested",
                "ext": ".svelte",
                "name": "foo",
                "path": "test/unit/filemapper/example/routes/no-layouts/nested/foo.svelte",
              },
              "id": "_default_no_layouts_nested_foo_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/routes/no-layouts/nested/foo.svelte",
              "name": "foo",
            },
            Object {
              "asyncModule": undefined,
              "children": Array [],
              "file": File {
                "base": "index.svelte",
                "dir": "test/unit/filemapper/example/routes/no-layouts/nested",
                "ext": ".svelte",
                "name": "index",
                "path": "test/unit/filemapper/example/routes/no-layouts/nested/index.svelte",
              },
              "id": "_default_no_layouts_nested_index_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/routes/no-layouts/nested/index.svelte",
              "name": "index",
            },
          ],
          "file": File {
            "base": "nested",
            "dir": "test/unit/filemapper/example/routes/no-layouts",
            "ext": "",
            "name": "nested",
            "path": "test/unit/filemapper/example/routes/no-layouts/nested",
          },
          "id": "_default_no_layouts_nested",
          "meta": Object {},
          "module": false,
          "name": "nested",
        },
      ],
      "file": File {
        "base": "no-layouts",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "no-layouts",
        "path": "test/unit/filemapper/example/routes/no-layouts",
      },
      "id": "_default_no_layouts",
      "meta": Object {},
      "module": false,
      "name": "no-layouts",
    },
  ],
  "file": File {
    "base": "routes",
    "dir": "test/unit/filemapper/example",
    "ext": "",
    "name": "routes",
    "path": "test/unit/filemapper/example/routes",
  },
  "id": "_default",
  "meta": Object {},
  "module": undefined,
  "name": "",
  "rootName": "default",
}
`;

exports[`test/unit/filemapper/filemapper.test.js/modules are merged with parent node: 2.filemap-with-modules 1`] = `
Object {
  "asyncModule": undefined,
  "children": Array [
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "_reset.svelte",
        "dir": "test/unit/filemapper/example/routes/admin",
        "ext": ".svelte",
        "name": "_reset",
        "path": "test/unit/filemapper/example/routes/admin/_reset.svelte",
      },
      "id": "_default_admin",
      "meta": Object {},
      "module": "test/unit/filemapper/example/routes/admin/_reset.svelte",
      "name": "admin",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "[slug].svelte",
            "dir": "test/unit/filemapper/example/routes/blog",
            "ext": ".svelte",
            "name": "[slug]",
            "path": "test/unit/filemapper/example/routes/blog/[slug].svelte",
          },
          "id": "_default_blog__slug__svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/routes/blog/[slug].svelte",
          "name": "[slug]",
        },
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "prefixed[slug].svelte",
            "dir": "test/unit/filemapper/example/routes/blog",
            "ext": ".svelte",
            "name": "prefixed[slug]",
            "path": "test/unit/filemapper/example/routes/blog/prefixed[slug].svelte",
          },
          "id": "_default_blog_prefixed_slug__svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/routes/blog/prefixed[slug].svelte",
          "name": "prefixed[slug]",
        },
      ],
      "file": File {
        "base": "blog",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "blog",
        "path": "test/unit/filemapper/example/routes/blog",
      },
      "id": "_default_blog",
      "meta": Object {},
      "module": false,
      "name": "blog",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "index.svelte",
        "dir": "test/unit/filemapper/example/routes",
        "ext": ".svelte",
        "name": "index",
        "path": "test/unit/filemapper/example/routes/index.svelte",
      },
      "id": "_default_index_svelte",
      "meta": Object {},
      "module": "test/unit/filemapper/example/routes/index.svelte",
      "name": "index",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [
            Object {
              "asyncModule": undefined,
              "children": Array [],
              "file": File {
                "base": "foo.svelte",
                "dir": "test/unit/filemapper/example/routes/no-layouts/nested",
                "ext": ".svelte",
                "name": "foo",
                "path": "test/unit/filemapper/example/routes/no-layouts/nested/foo.svelte",
              },
              "id": "_default_no_layouts_nested_foo_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/routes/no-layouts/nested/foo.svelte",
              "name": "foo",
            },
            Object {
              "asyncModule": undefined,
              "children": Array [],
              "file": File {
                "base": "index.svelte",
                "dir": "test/unit/filemapper/example/routes/no-layouts/nested",
                "ext": ".svelte",
                "name": "index",
                "path": "test/unit/filemapper/example/routes/no-layouts/nested/index.svelte",
              },
              "id": "_default_no_layouts_nested_index_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/routes/no-layouts/nested/index.svelte",
              "name": "index",
            },
          ],
          "file": File {
            "base": "nested",
            "dir": "test/unit/filemapper/example/routes/no-layouts",
            "ext": "",
            "name": "nested",
            "path": "test/unit/filemapper/example/routes/no-layouts/nested",
          },
          "id": "_default_no_layouts_nested",
          "meta": Object {},
          "module": false,
          "name": "nested",
        },
      ],
      "file": File {
        "base": "no-layouts",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "no-layouts",
        "path": "test/unit/filemapper/example/routes/no-layouts",
      },
      "id": "_default_no_layouts",
      "meta": Object {},
      "module": false,
      "name": "no-layouts",
    },
  ],
  "file": File {
    "base": "_module.svelte",
    "dir": "test/unit/filemapper/example/routes",
    "ext": ".svelte",
    "name": "_module",
    "path": "test/unit/filemapper/example/routes/_module.svelte",
  },
  "id": "_default",
  "meta": Object {},
  "module": "test/unit/filemapper/example/routes/_module.svelte",
  "name": "",
  "rootName": "default",
}
`;

exports[`test/unit/filemapper/filemapper.test.js/options get added: 3.filemap-with-resets 1`] = `
Object {
  "asyncModule": undefined,
  "children": Array [
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "_reset.svelte",
        "dir": "test/unit/filemapper/example/routes/admin",
        "ext": ".svelte",
        "name": "_reset",
        "path": "test/unit/filemapper/example/routes/admin/_reset.svelte",
      },
      "id": "_default_admin",
      "meta": Object {
        "reset": true,
      },
      "module": "test/unit/filemapper/example/routes/admin/_reset.svelte",
      "name": "admin",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "[slug].svelte",
            "dir": "test/unit/filemapper/example/routes/blog",
            "ext": ".svelte",
            "name": "[slug]",
            "path": "test/unit/filemapper/example/routes/blog/[slug].svelte",
          },
          "id": "_default_blog__slug__svelte",
          "meta": Object {
            "dynamic": true,
            "order": false,
          },
          "module": "test/unit/filemapper/example/routes/blog/[slug].svelte",
          "name": "[slug]",
        },
        Object {
          "asyncModule": undefined,
          "children": Array [],
          "file": File {
            "base": "prefixed[slug].svelte",
            "dir": "test/unit/filemapper/example/routes/blog",
            "ext": ".svelte",
            "name": "prefixed[slug]",
            "path": "test/unit/filemapper/example/routes/blog/prefixed[slug].svelte",
          },
          "id": "_default_blog_prefixed_slug__svelte",
          "meta": Object {
            "dynamic": true,
            "order": false,
          },
          "module": "test/unit/filemapper/example/routes/blog/prefixed[slug].svelte",
          "name": "prefixed[slug]",
        },
      ],
      "file": File {
        "base": "blog",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "blog",
        "path": "test/unit/filemapper/example/routes/blog",
      },
      "id": "_default_blog",
      "meta": Object {},
      "module": false,
      "name": "blog",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [],
      "file": File {
        "base": "index.svelte",
        "dir": "test/unit/filemapper/example/routes",
        "ext": ".svelte",
        "name": "index",
        "path": "test/unit/filemapper/example/routes/index.svelte",
      },
      "id": "_default_index_svelte",
      "meta": Object {
        "isDefault": true,
      },
      "module": "test/unit/filemapper/example/routes/index.svelte",
      "name": "index",
    },
    Object {
      "asyncModule": undefined,
      "children": Array [
        Object {
          "asyncModule": undefined,
          "children": Array [
            Object {
              "asyncModule": undefined,
              "children": Array [],
              "file": File {
                "base": "foo.svelte",
                "dir": "test/unit/filemapper/example/routes/no-layouts/nested",
                "ext": ".svelte",
                "name": "foo",
                "path": "test/unit/filemapper/example/routes/no-layouts/nested/foo.svelte",
              },
              "id": "_default_no_layouts_nested_foo_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/routes/no-layouts/nested/foo.svelte",
              "name": "foo",
            },
            Object {
              "asyncModule": undefined,
              "children": Array [],
              "file": File {
                "base": "index.svelte",
                "dir": "test/unit/filemapper/example/routes/no-layouts/nested",
                "ext": ".svelte",
                "name": "index",
                "path": "test/unit/filemapper/example/routes/no-layouts/nested/index.svelte",
              },
              "id": "_default_no_layouts_nested_index_svelte",
              "meta": Object {
                "isDefault": true,
              },
              "module": "test/unit/filemapper/example/routes/no-layouts/nested/index.svelte",
              "name": "index",
            },
          ],
          "file": File {
            "base": "nested",
            "dir": "test/unit/filemapper/example/routes/no-layouts",
            "ext": "",
            "name": "nested",
            "path": "test/unit/filemapper/example/routes/no-layouts/nested",
          },
          "id": "_default_no_layouts_nested",
          "meta": Object {},
          "module": false,
          "name": "nested",
        },
      ],
      "file": File {
        "base": "no-layouts",
        "dir": "test/unit/filemapper/example/routes",
        "ext": "",
        "name": "no-layouts",
        "path": "test/unit/filemapper/example/routes/no-layouts",
      },
      "id": "_default_no_layouts",
      "meta": Object {},
      "module": false,
      "name": "no-layouts",
    },
  ],
  "file": File {
    "base": "_module.svelte",
    "dir": "test/unit/filemapper/example/routes",
    "ext": ".svelte",
    "name": "_module",
    "path": "test/unit/filemapper/example/routes/_module.svelte",
  },
  "id": "_default",
  "meta": Object {},
  "module": "test/unit/filemapper/example/routes/_module.svelte",
  "name": "",
  "rootName": "default",
}
`;
